File: Core/Components/RichTextBox.cs
using Bridge.Html5;
using Core.Clients;
using Core.Extensions;
using Core.Models;
using Core.MVVM;
using Core.Structs;
using System;
using System.Collections.Generic;

namespace Core.Components
{
    public class RichTextBox : EditableComponent
    {
        private readonly object editor;

        public RichTextBox(Component ui, HTMLElement ele = null) : base(ui)
        {
            if (Meta.Row <= 0)
            {
                Meta.Row = 1;
            }
            if (ele != null)
            {
                ParentElement = ele;
                BindingWebComponent();
            }
            else
            {
                ParentElement = ParentElement ?? Html.Context;
                BindingWebComponent();
            }
            ParentElement.AppendChild(Element);
        }

        private void BindingWebComponent()
        {
            Element = Html.Take(ParentElement).Div.Id(Uuid7.Id25()).GetContext();
        }

        public override void Render()
        {
            var self = this;
            /*@
            initCkEditor(self);
            */
        }

        public override void UpdateView(bool force = false, bool? dirty = null, params string[] componentNames)
        {
            var handler = _events.GetValueOrDefault(nameof(UpdateView));
            handler?.Invoke(new ObservableArgs { Com = this, EvType = EventType.Change });
        }
    }
}

File: CoreAPI/wwwRoot/js/core/richTextBox.js
import EditableComponent from "./editableComponent";
import { Html } from "./utils/html.js";
import EventType from './models/eventType.js';
import ObservableArgs from './models/observable.js';
import { Uuid7 } from "./structs/uuidv7.js";
import { Component } from "models/component.js";

import "./utils/fix.js";


export class RichTextBox extends EditableComponent {
     /**
     * @param {Component} ui
     * @param {HTMLElement} [ele=null] 
     */
    constructor(ui, ele = null) {
        super(ui,ele);
        this.defaultValue = "";
        if (this.Meta.Row <= 0) {
                this.Meta.Row = 1;
        }
        if (ele != null)
            {
                this.ParentElement = ele ; 
                this.BindingWebComponent();
            }
            else
            {
                this.ParentElement = this.ParentElement ?? Html.Context;
                this.BindingWebComponent();
            }
        
            this.ParentElement.appendChild(this.Element);
    }

    BindingWebComponent() {
        this.Element = Html.Take(this.ParentElement).Div.Id(Uuid7.Id25()).GetContext();
    }

    Render() {
        initCkEditor(this);
    }

    UpdateView(force = false, dirty = null, ...componentNames) {
        const handler = this._events["UpdateView"];
        if (handler) {
            const args = new ObservableArgs();
            args.Com = this;
            args.EvType = 'Change';
            handler(args);
        }
    }
}
